File: 03423. Maximum Difference Between Adjacent Elements in a Circular Array/solution.go
/**
 * Problem: 3423. Maximum Difference Between Adjacent Elements in a Circular Array
 *
 * Difficulty: Easy
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func abs(x int) int {
	if x < 0 {
		return -x
	}

	return x
}

func maxAdjacentDistance(nums []int) int {
	n := len(nums)
	// Initialize max difference with the circular connection (last to first element)
	maxDiff := abs(nums[0] - nums[n-1])

	// Check all adjacent pairs in the array
	for i := 0; i < n-1; i++ {
		maxDiff = max(maxDiff, abs(nums[i]-nums[i+1]))
	}

	return maxDiff
}

File: 03423. Maximum Difference Between Adjacent Elements in a Circular Array/solution.js
/**
 * Problem: 3423. Maximum Difference Between Adjacent Elements in a Circular Array
 *
 * Difficulty: Easy
 *
 * Language: JavaScript
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

/**
 * Find max absolute difference between adjacent elements in circular array
 *
 * @param {number[]} nums - Circular array of integers
 *
 * @returns {number} - Max absolute difference
 */
const maxAdjacentDistance = (nums) => {
  const n = nums.length
  // Initialize max difference with the circular connection (last to first element)
  let maxDiff = Math.abs(nums[0] - nums[n - 1])

  // Check all adjacent pairs in the array
  for (let i = 0; i < n - 1; i++)
    maxDiff = Math.max(maxDiff, Math.abs(nums[i] - nums[i + 1]))

  return maxDiff
}
